qa/tasks/kafka.py

```diff
@@ -0,0 +1,401 @@
+    log.info('Downloading bucket-notification-tests...')
+    testdir = teuthology.get_testdir(ctx)
+    for (client, client_config) in config.items():
+        bntests_branch = client_config.get('force-branch', None)
+        if not bntests_branch:
+            raise ValueError(
+                "Could not determine what branch to use for bn-tests. Please add 'force-branch: {bn-tests branch name}' to the .yaml config for this bucket notifications tests task.")
+
+        log.info("Using branch '%s' for bucket notifications tests", bntests_branch)
+        sha1 = client_config.get('sha1')
+        git_remote = client_config.get('git_remote', teuth_config.ceph_git_base_url)
+        ctx.cluster.only(client).run(
+            args=[
+                'git', 'clone',
+                '-b', bntests_branch,
+                git_remote + 'ceph.git',
+                '{tdir}/ceph'.format(tdir=testdir),
+                ],
+            )
+        if sha1 is not None:
+            ctx.cluster.only(client).run(
+                args=[
+                    'cd', '{tdir}/ceph'.format(tdir=testdir),
+                    run.Raw('&&'),
+                    'git', 'reset', '--hard', sha1,
+                    ],
+                )
+    try:
+        yield
+    finally:
+        log.info('Removing bucket-notifications-tests...')
+        testdir = teuthology.get_testdir(ctx)
+        for client in config:
+            ctx.cluster.only(client).run(
+                args=[
+                    'rm',
+                    '-rf',
+                    '{tdir}/ceph'.format(tdir=testdir),
+                    ],
+                )
+
+@contextlib.contextmanager
+def build(ctx, config):
+    """
+    Building ceph to get the build directory
+    """
+    assert isinstance(config, dict)
+    log.info('Building ceph...')
+    testdir = teuthology.get_testdir(ctx)
+    for (client, _) in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+        ctx.cluster.only(client).run(args=['cd', '{tdir}/ceph'.format(tdir=testdir), run.Raw('&&'), './install-deps.sh'])
+        ctx.cluster.only(client).run(args=['cd', '{tdir}/ceph'.format(tdir=testdir), run.Raw('&&'), './do_cmake.sh', '-DWITH_MGR_DASHBOARD_FRONTEND=OFF', '-DWITH_KRBD=OFF'])
+        ctx.cluster.only(client).run(args=['cd', '{tdir}/ceph/build'.format(tdir=testdir), run.Raw('&&'), 'make', '-j', '4'])
+
+    try:
+        yield
+    finally:
+        pass
+
+@contextlib.contextmanager
+def install_kafka(ctx, config):
+    """
+    Downloading the kafka tar file.
+    """
+    assert isinstance(config, dict)
+    log.info('Installing Kafka...')
+
+    for (client, _) in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+        test_dir=teuthology.get_testdir(ctx)
+        toxvenv_sh(ctx, remote, ['cd', '{tdir}/ceph'.format(tdir=test_dir), run.Raw('&&'), 'wget', 'https://mirrors.estointernet.in/apache/kafka/2.6.0/kafka-2.6.0-src.tgz'])
```

Comment:
better not to hardcode the Indian mirror. would be slow for the sepia servers

Reply:
Okay, so can you please suggest an alternative?

Reply:
from official site: https://kafka.apache.org/downloads

Reply:
Cool